src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -22,6 +22,15 @@ public ShoppingCart create(ShoppingCart shoppingCart) {
                 .findFirst();
     }
 
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return getAll().stream()
+                .filter(shoppingCart -> shoppingCart.getUserId().equals(userId))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(
```

Comment:
let's return optional at the dao layer and get the object at the service layer)

Reply:
Agree, fixed